### PR TITLE
:tada: Render wasm fill images

### PR DESCRIPTION
**Pull request by superalex**
Related to https://tree.taiga.io/project/penpot/task/9268: 

https://github.com/user-attachments/assets/12e07af5-48a0-47e8-a6db-7352c30b5d40

### CODE REVIEW

File: render-wasm/src/render.rs [hunk 1]
```diff
@@ -157,9 +161,46 @@ impl RenderState {
...
+                //           let b64_image = images::fetch_base_64_image(&url);
+                //           let image_bytes = BASE64_STANDARD.decode(b64_image).expect("Error decoding Base64");
+                //           let image_data = unsafe { skia::Data::new_bytes(&image_bytes) };
+                //           let image = images::Image::from_encoded(image_data).expect("Error loading image");
+                //           self.images.insert(url.clone(), image.clone());
+                //           images::draw_image_contain(& self.drawing_surface.canvas(), &image, image_fill.width, image_fill.height, shape.selrect);
+                //       }                
+                // }
+                let image = self.images.get(&url).cloned().or_else(|| {
+                    let b64_image = images::fetch_base_64_image(&url);
```
Comment (superalex) [inline render-wasm/src/render.rs]: We could have the data in bytes directly

File: render-wasm/src/images.rs [hunk 2]
```diff
@@ -1,3 +1,51 @@
+use crate::math;
```
Comment (belen-albeza) [inline render-wasm/src/images.rs]: IMO this `images.rs` should be a submodule of `shapes`

File: render-wasm/src/images.rs [hunk 3]
```diff
@@ -1,3 +1,51 @@
...
 use skia_safe as skia;
 
 pub type Image = skia::Image;
+
+pub fn draw_image_contain(
```
Comment (belen-albeza) [inline render-wasm/src/images.rs]: I don't understand this function name (maybe `draw_image_in_container`?

File: render-wasm/src/images.rs [hunk 4]
```diff
@@ -1,3 +1,51 @@
...
+    canvas: &skia::Canvas,
+    image: &Image,
+    width: f32,
+    height: f32,
```
Comment (belen-albeza) [inline render-wasm/src/images.rs]: let's group this params into a tuple `(f32, f32)`.

File: render-wasm/src/main.rs [hunk 5]
```diff
@@ -208,6 +208,59 @@ pub extern "C" fn add_shape_fill_stops(ptr: *mut RawStopData, n_stops: i32) {
     }
 }
 
+#[no_mangle]
+pub extern "C" fn store_image(a: u32, b: u32, c: u32, d: u32, ptr: *mut u8, size: i32) {
+    if ptr.is_null() || size == 0 {
+        panic!("Datos inválidos: puntero nulo o tamaño cero");
```
Comment (belen-albeza) [inline render-wasm/src/main.rs]: English

File: render-wasm/src/main.rs [hunk 6]
```diff
@@ -208,6 +208,59 @@ pub extern "C" fn add_shape_fill_stops(ptr: *mut RawStopData, n_stops: i32) {
     }
 }
 
+#[no_mangle]
+pub extern "C" fn store_image(a: u32, b: u32, c: u32, d: u32, ptr: *mut u8, size: i32) {
```
Comment (belen-albeza) [inline render-wasm/src/main.rs]: `size` should be `u32`

**Pull request merged**